docs/existing-aws-infrastructure.md

```diff
@@ -0,0 +1,94 @@
+
+1. The ID of the VPC
+2. The IDs of the subnets in the VPC that you want Cluster API to use
+
+Note that there is no need to create an Elastic Load Balancer (ELB), security groups, or EC2 instances; Cluster API will take care of these items.
+
+## Tagging AWS Resources
+
+In order for Cluster API to properly recognize and consume existing AWS resources, these resources must be tagged with a specific set of tags. The list below provides the tags that must be present; in this list, `<cluster-name>` refers to the name of cluster as specified in the `metadata.name` field of the Cluster object (or manifest). Here are the tags that are required:
+
+    sigs.k8s.io/cluster-api-provider-aws/cluster/<cluster-name>
+    sigs.k8s.io/cluster-api-provider-aws/role
+
+These two tags should be present on the following AWS resources:
+
+* VPC
+* All subnets that Cluster API will utilize (private and public)
+* Route tables
+* Internet gateway
+* NAT gateways
+* Elastic IPs assigned to NAT gateways
+
+For the `sigs.k8s.io/cluster-api-provider-aws/cluster/<cluster-name>` tag, a value of "owned" is used by Cluster API itself to indicate resources are owned by a specific cluster, and that the lifecycle of the resource is tied to the lifecycle of the cluster. Cluster API recognizes a value of "shared" to indicate a resource may be shared between multiple clusters, and the lifecycle of said resource is not tied to any particular cluster. In this situation---where AWS infrastructure is _not_ being managed by Cluster API---the value should be something **other** than "owned". Using "shared" is acceptable.
+
+For the `sigs.k8s.io/cluster-api-provider-aws/role` tag, Cluster API uses a variety of values to indicate resources are associated with a particular role. Here are the tag values that should be used:
+
+* VPC, route tables, Internet gateway, and NAT gateways should use the value "common".
+* Public subnets should use the value "public".
+* Private subnets should use the value "private".
+* Elastic IPs (for use by the NAT gateways) should use the value "apiserver".
+
+Additionally, for proper operation of the AWS cloud provider, all subnets where Kubernetes nodes reside should also have the `kubernetes.io/cluster/<cluster-name>` tag present.
+
+## Configuring the AWSCluster Specification
+
+Specifying existing infrastructure for Cluster API to use takes place in the specification for the AWSCluster object. Specifically, you will need to add an entry with the VPC ID and the IDs of all applicable subnets into the `networkSpec` field. Here is an example:
+
+```yaml
+spec:
+  networkSpec:
+    vpc:
+      id: vpc-0425c335226437144
+    subnets:
+      - id: subnet-07758a9bc904d06af
+      - id: subnet-0a3507a5ad2c5c8c3
+      - id: subnet-02ad6429dd0532452
+      - id: subnet-02b300779e9d895cf
+      - id: subnet-03d8f353b289b025f
+      - id: subnet-0a2fe03b0d88fa078
```

Comment:
As mentioned above, it's not necessary to include the subnets here. However, we do find it useful to include a default `sshKeyName` and `region` at this point, like so:

```
spec:
  networkSpec:
    vpc:
      id: vpc-0425c335226437144
  region: us-west-2
  sshKeyName: some-ssh-key
```

Though neither are required.

Reply:
Removed references to the list of subnets via 38ce860.